nucypher/blockchain/eth/sol/source/contracts/PREApplication.sol

```diff
@@ -0,0 +1,634 @@
+                (rewardPerToken() - info.rewardPerTokenPaid)
+                / 1e18
+                + info.tReward;
+        return result.toUint96();
+    }
+
+    /**
+    * @notice Transfer reward for the next period. Can be called only by distributor
+    * @param _reward Amount of reward
+    */
+    function pushReward(uint96 _reward) external updateReward(address(0)) {
+        require(msg.sender == rewardDistributor, "Only distributor can transfer reward");
+        require(_reward > 0, "Reward must be specified");
+        if (block.timestamp >= periodFinish) {
+            rewardRate = (_reward / rewardDuration).toUint96();
+        } else {
+            uint256 remaining = periodFinish - block.timestamp;
+            uint256 leftover = remaining * rewardRate;
+            rewardRate = ((_reward + leftover) / rewardDuration).toUint96();
+        }
+        lastUpdateTime = block.timestamp;
+        periodFinish = block.timestamp + rewardDuration;
+        emit RewardAdded(_reward);
+        token.safeTransferFrom(msg.sender, address(this), _reward);
+    }
+
+    /**
+    * @notice Withdraw available amount of T reward to beneficiary. Can be called only by beneficiary
+    * @param _operator Operator address
+    */
+    function withdraw(address _operator) external updateReward(_operator) {
+        address beneficiary = getBeneficiary(_operator);
+        require(msg.sender == beneficiary, "Caller must be beneficiary");
+
+        OperatorInfo storage info = operatorInfo[_operator];
+        require(info.tReward > 0, "No reward to withdraw");
+        uint96 value = info.tReward;
+        info.tReward = 0;
+        emit RewardPaid(_operator, beneficiary, value);
+        token.safeTransfer(beneficiary, value);
+    }
+
+    //------------------------Authorization------------------------------
+
+    /**
+    * @notice Recalculate reward and save increased authorization. Can be called only by staking contract
+    * @param _operator Address of operator
+    * @param _fromAmount Amount of previously authorized tokens to PRE application by operator
+    * @param _toAmount Amount of authorized tokens to PRE application by operator
+    */
+    function authorizationIncreased(
+        address _operator,
+        uint96 _fromAmount,
+        uint96 _toAmount
+    )
+        external override onlyStakingContract updateReward(_operator)
+    {
+        require(_operator != address(0) && _toAmount > 0, "Input parameters must be specified");
+        require(_toAmount >= minAuthorization, "Authorization must be greater than minimum");
+
+        OperatorInfo storage info = operatorInfo[_operator];
+        require(
+            _operatorFromWorker[_operator] == address(0) ||
+            _operatorFromWorker[_operator] == info.worker,
+            "An operator can't be a worker for another operator"
+        );
+
+        if (info.workerConfirmed) {
+            authorizedOverall += _toAmount - _fromAmount;
```

Comment:
App should do everything to execute method so `require` is too strong but you're right!, if `_fromAmount` is not equal `authorized` then it will lead to error. I'll cahnge it to `_toAmount - info.authorized` and will make additional tests

Reply:
✔️